raft/src/main/java/org/apache/kafka/raft/UnattachedState.java

```diff
@@ -68,7 +73,9 @@ public UnattachedState(
 
     @Override
     public ElectionState election() {
-        if (leaderId.isPresent()) {
+        if (votedState.isPresent()) {
+            return votedState.get().election();
```

Comment:
I would remove `VotedState#election` and move the body of the method here. Future readers can see all of the returned election state by just looking at this method.